hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/mapreduce/MapReduceRestoreToOriginalSplitsJob.java

```diff
@@ -66,41 +73,23 @@ public Configuration getConf() {
     return conf;
   }
 
-  private static Map<byte[], List<Path>> buildFamily2Files(FileSystem fs, Path[] dirs,
-    boolean isFullBackup) throws IOException {
-    if (isFullBackup) {
-      return buildFullBackupFamily2Files(fs, dirs);
+  private static DistCp getDistCp(Configuration conf, boolean fullBackupRestore) throws Exception {
+    if (fullBackupRestore) {
+      return new DistCp(conf, null);
     }
 
-    Map<byte[], List<Path>> family2Files = new HashMap<>();
+    return new IncrementalBackupDistCp(conf);
+  }
 
-    for (Path dir : dirs) {
-      byte[] familyName = Bytes.toBytes(dir.getParent().getName());
-      if (family2Files.containsKey(familyName)) {
-        family2Files.get(familyName).add(dir);
-      } else {
-        family2Files.put(familyName, Lists.newArrayList(dir));
-      }
+  private static class IncrementalBackupDistCp extends DistCp {
+    public IncrementalBackupDistCp(Configuration conf) throws Exception {
+      super(conf, null);
     }
 
-    return family2Files;
-  }
-
-  private static Map<byte[], List<Path>> buildFullBackupFamily2Files(FileSystem fs, Path[] dirs)
-    throws IOException {
-    Map<byte[], List<Path>> family2Files = new HashMap<>();
-    for (Path regionPath : dirs) {
-      FSVisitor.visitRegionStoreFiles(fs, regionPath, (region, family, name) -> {
-        Path path = new Path(regionPath, new Path(family, name));
-        byte[] familyName = Bytes.toBytes(family);
-        if (family2Files.containsKey(familyName)) {
-          family2Files.get(familyName).add(path);
-        } else {
-          family2Files.put(familyName, Lists.newArrayList(path));
-        }
-      });
+    @Override
+    protected Path createInputFileListing(Job job) throws IOException {
```

Comment:
We need to do the same thing we're currently doing [here](https://github.com/apache/hbase/blob/e56e14bcc9f15ceae3f7961a21d4eef94ce1c159/hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/IncrementalTableBackupClient.java#L435C52-L435C57) or else the subsequent bulkload doesn't work correctly, specifically for incremental backups